### ISSUE TITLE
[accessibility] The map container should be a programmatically identifiable element

### ISSUE THREAD

**Issue by Malvoz** [openlayers/openlayers#12314]
To give visually impaired users context for the map as a whole, when authors provide the following HTML:

```html
<div id="map" class="map"></div>
```
it should produce:
```html
<div id="map" class="map" role="region" aria-roledescription="map"></div>
```

<span title="Web Accessibility Initiative - Accessible Rich Internet Applications">WAI-ARIA<span> definitions:
- [`role="region"`](https://www.w3.org/TR/wai-aria-1.1/#region)
- [`aria-roledescription`](https://www.w3.org/TR/wai-aria-1.1/#aria-roledescription)

<hr>

This is essentially implementing the WCAG 2.1 [technique ARIA11](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA11) which satisfies Success Criterion [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships), [1.3.6 Identify Purpose](https://www.w3.org/WAI/WCAG21/Understanding/identify-purpose) and [2.4.1 Bypass Blocks](https://www.w3.org/WAI/WCAG21/Understanding/bypass-blocks) for the map container.

Other related resources:

- https://www.scottohara.me/blog/2018/03/03/landmarks.html
- https://www.w3.org/TR/wai-aria-practices/examples/landmarks/
- https://www.w3.org/TR/wai-aria-practices/examples/landmarks/region.html
